src/index.ts

```diff
@@ -92,35 +100,50 @@ map.on('style.load', () => {
   animationService.readdRunToMap(currentRun);
 });
 
+function showRunButtons(): void {
+  removeLastElement.classList.remove('slide-out');
+  removeLastElement.classList.add('slide-in');
+  removeLastElement.setAttribute('aria-hidden', 'false');
+  saveRunElement.classList.remove('hidden');
+  clearRunElement.classList.remove('hidden');
+}
+
+function hideRunButtons(): void {
+  removeLastElement.classList.remove('slide-in');
+  removeLastElement.classList.add('slide-out');
+  removeLastElement.setAttribute('aria-hidden', 'true');
+  saveRunElement.classList.add('hidden');
+  clearRunElement.classList.add('hidden');
+}
+
 function addNewPoint(e: MapMouseEvent): void {
   if (currentRun === undefined) {
     let start = new RunStart(
       e.lngLat
     );
     start.setMarker(addMarker(e.lngLat, true));
     currentRun = new CurrentRun(start);
-    removeLastElement.classList.remove('slide-out');
-    removeLastElement.classList.add('slide-in');
-    removeLastElement.setAttribute('aria-hidden', 'false');
+    showRunButtons();
     updateLengthElement();
   } else {
     let prev = currentRun.getLastPosition();
     if (followRoads) {
-      addSegmentFromDirectionsResponse(prev, e);
+      addSegmentFromDirectionsResponse(prev, e.lngLat);
     } else {
-      addSegmentFromStraightLine(prev, e);
+      addSegmentFromStraightLine(prev, e.lngLat);
     }
   }
   setWaiting(false);
+  setTimeout(() => preferenceService.saveLastRun(runToJson(currentRun)), 100); // for some reason this won't save right without a delay
```

Comment:
good to know, I'll have to play around with this some more and see what I can find out 👍